preupg/application.py

```diff
@@ -489,13 +493,10 @@ def scan_system(self):
                 return ReturnValues.SCENARIO
             self.report_parser.modify_platform_tag(version[0])
         if self.conf.mode:
-            try:
-                lines = [i.rstrip() for i in FileHelper.get_file_content(os.path.join(self.assessment_dir,
-                                                                                      self.conf.mode),
-                                                                         'rb',
-                                                                         method=True)]
-            except IOError:
-                return
+            lines = [i.rstrip() for i in FileHelper.get_file_content(os.path.join(self.assessment_dir,
+                                                                                  self.conf.mode),
+                                                                     'rb',
+                                                                     method=True)]
```

Comment:
TL;DR: Basically, this was just a poor error handling.

If user uses the --mode option, only the appropriate modules are supposed to be run (either for upgrade or migration). And for that two files exist - _/root/preupgrade/RHEL5_7/{migrate,upgrate}_ - holding a list of modules to run. However if there is any error with reading this file, the scan_system function just returns None and no further assessment happens. User (and then also developer who would troubleshoot this) is left with no information whatsoever what went wrong.